### DISCUSSION TITLE
Modify AirSim Plugins of released environments

### DISCUSSION THREAD

**Discussion by joigalcar3** [microsoft/AirSim#3845] [Support Q&A]
Hi everyone! 

With every new release, AirSim provides environments that can be used out of the box with Airsim, such as the AirSimNH and the CityEnviron ([AirSim releases](https://github.com/Microsoft/AirSim/releases)). They usually come with an executable and an .uproject file. 

During the last months, I have added some functions to the AirSim API within the Blocks environment, changes which are reflected in the Plugins folder that has to be added to the folder of an environment. At this point, I will like to run my new code in another environment in order to gather some data (namely the CityEnviron). Is there a way to modify the AirSim plugins that the released environments use or is that only possible when the environment has been purchased from the Marketplace?

So far I have tried to follow the [Unreal AirSim Setup from scratch]{https://www.youtube.com/watch?v=1oY8Qu5maQQ} tutorial with multiple environments, including the AirSim released Landscape Mountains with no success. Every trial with the .uproject file has lead to the following window:

![modules_missing](https://user-images.githubusercontent.com/31962886/124253496-5ec3d800-db28-11eb-823a-bfdf13302feb.PNG)

After clicking "Yes", UE4 announces that it starts to build and is followed by this next window: 

![error](https://user-images.githubusercontent.com/31962886/124253641-861aa500-db28-11eb-80df-d23f5981c9ff.PNG)

Therefore, I do not know whether changing the plugins of the released environments is possible at all. If that is not the case, then what is the purpose of the .uproject file of the released environments? Is the .exe file the only one that can be used?

Sorry for the lengthy explanation, but I wanted to be as clear as possible since I have been trying to solve this problem for a while. 

Thanks a lot for your help and time!!

**Comment by zimmy87** [comment]
The .uproject file is an artifact of the build process and can't be opened in the editor regularly as there are no corresponding source files for it. Currently, the only way to include an updated version of the plugin in an environment is to recompile the environment from source. Unfortunately, we don't have the licenses to distribute the source of the release environments.

**Reply by joigalcar3** [reply]
@zimmy87 Thanks a lot for your help and sorry for my late response! From your answer, I understand that the environments must be acquired from the Marketplace in order to use the modified AirSim library. Furthermore, I was wondering whether the .uproject file of the released environments has any use or only the .exe file should be considered for any application. Again, thanks a lot for your time!
